core/src/main/java/io/questdb/tasks/TelemetryWalTask.java

```diff
@@ -0,0 +1,110 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.tasks;
+
+import io.questdb.Telemetry;
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.TableWriter;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.ObjectFactory;
+import org.jetbrains.annotations.NotNull;
+
+public class TelemetryWalTask extends AbstractTelemetryTask {
+    public static final String TABLE_NAME = "telemetry_wal";
+
+    private static final Log LOG = LogFactory.getLog(TelemetryWalTask.class);
+
+    private short event;
+    private int tableId;
+    private int walId;
+    private long seqTxn;
+    private long rowCount;
+    private long latency; //micros
+
+    private TelemetryWalTask() {
+    }
+
+    public static long store(@NotNull Telemetry<TelemetryWalTask> telemetry, short origin, short event, int tableId, int walId, long seqTxn, long rowCount, long start) {
+        final TelemetryWalTask task = telemetry.nextTask();
+        if (task != null) {
+            task.origin = origin;
+            task.event = event;
+            task.tableId = tableId;
+            task.walId = walId;
+            task.seqTxn = seqTxn;
+            task.rowCount = rowCount;
+            task.latency = task.created - start;
+            telemetry.store();
+            return task.created;
+        }
+        return -1L;
+    }
+
+    @Override
+    public void writeTo(TableWriter writer) {
+        try {
+            final TableWriter.Row row = writer.newRow(created);
+            row.putShort(1, event);
+            row.putShort(2, origin);
+            row.putInt(3, tableId);
+            row.putInt(4, walId);
+            row.putLong(5, seqTxn);
+            row.putLong(6, rowCount);
+            row.putLong(7, latency);
+            row.append();
+        } catch (CairoException e) {
+            LOG.error().$("Could not insert a new ").$(TABLE_NAME).$(" row [errno=").$(e.getErrno())
+                    .$(", error=").$(e.getFlyweightMessage())
+                    .$(']').$();
+        }
+    }
+
+    public static final Telemetry.TelemetryType<TelemetryWalTask> TYPE = new Telemetry.TelemetryType<TelemetryWalTask>() {
+        @Override
+        public String getTableName() {
+            return TABLE_NAME;
+        }
+
+        @Override
+        public String getCreateSql() {
+            return "CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " (" +
+                    "created timestamp, " +
```

Comment:
I'd add transaction count which always equals to 1 now so that we can compress multiple transactions in 1 row in the future (resample).

Reply:
`txn count` - this is extra storage, let's add it when we have this working.

QuestDB version is already saved into the `telemetry_config` table.
it is probably not worth to repeat it on every line in WAL telemetry.